soundcard/pulseaudio.py

```diff
@@ -333,29 +332,17 @@ class _Recorder(_Stream):
 
     """
 
+    def __init__(self, *args, **kwargs):
+        super(_Recorder, self).__init__(*args, **kwargs)
+        self._pending_chunk = numpy.zeros((0, ))
+
     def _connect_stream(self, bufattr):
         self._pulse._pa_stream_connect_record(self.stream, self._id.encode(), bufattr, _pa.PA_STREAM_ADJUST_LATENCY)
 
-    def record(self, numframes):
-        """Record some audio data.
-
-        The data will be returned as a `frames × channels` float32
-        numpy array.
-
-        This function will wait until `numframes` frames have been
-        recorded. However, the audio backend holds the final authority
-        over how much audio data can be read at a time, so the
-        returned amount of data will often be slightly larger than
-        what was requested. The amount of buffering can be controlled
-        through the blocksize of the recorder object.
-
-        """
-
-        captured_frames = 0
-        captured_data = []
+    def _record_chunk(self):
         data_ptr = _ffi.new('void**')
         nbytes_ptr = _ffi.new('size_t*')
-        while captured_frames < numframes:
+        while 1:
```

Comment:
should be `while True`

Reply:
I started with `while True` but read that `while 1` is faster. 
Now it's clearer, `while 1` was faster in python 2.x because Python needed to do a lookup for `True` (because you could redefine in) but now `True` is a constant so it's the same. I'll rewrite with `while True`.